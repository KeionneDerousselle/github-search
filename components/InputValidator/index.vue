<template>
  <ValidationProvider
    v-slot="{ errors, failedRules, aria, classes, validate, reset, valid, invalid, changed, touched, untouched, pristine, dirty, pending, required, validated, passed, failed }"
    :vid="vid"
    :tag="tag"
    :skip-if-empty="skipIfEmpty"
    :rules="rules"
    :name="name || label"
    :mode="mode"
    :immediate="immediate"
    :disabled="disabled"
    :debounce="debounce"
    :custom-messages="customMessages"
    :bails="bails"
    class="input">
    <label
      v-if="label"
      :for="labelFor"
      class="input__label sr-only">
      {{ label }}
    </label>

    <slot
      :errors="errors"
      :failedRules="failedRules"
      :aria="aria"
      :classes="classes"
      :validate="validate"
      :reset="reset"
      :valid="valid"
      :invalid="invalid"
      :changed="changed"
      :touched="touched"
      :untouched="untouched"
      :pristine="pristine"
      :dirty="dirty"
      :pending="pending"
      :required="required"
      :validated="validated"
      :passed="passed"
      :failed="failed" />

    <slot
      v-if="!errors || errors.length < 1"
      name="hint"
      :errors="errors"
      :failedRules="failedRules"
      :aria="aria"
      :classes="classes"
      :validate="validate"
      :reset="reset"
      :valid="valid"
      :invalid="invalid"
      :changed="changed"
      :touched="touched"
      :untouched="untouched"
      :pristine="pristine"
      :dirty="dirty"
      :pending="pending"
      :required="required"
      :validated="validated"
      :passed="passed"
      :failed="failed">
      <div
        v-if="hint"
        class="input__hint-container">
        <span class="input__hint text-black-500">{{ hint }}</span>
      </div>
    </slot>

    <slot
      v-if="showErrors && errors && errors.length > 0"
      name="errors"
      :errors="errors"
      :failedRules="failedRules"
      :classes="classes"
      :valid="valid"
      :invalid="invalid"
      :changed="changed"
      :touched="touched"
      :untouched="untouched"
      :pristine="pristine"
      :dirty="dirty"
      :pending="pending"
      :required="required"
      :validated="validated"
      :passed="passed"
      :failed="failed">
      <div class="input__errors">
        <span
          v-if="bails"
          class="input__error text-error-400">
          {{ errors[0] }}
        </span>
        <span
          v-for="error in errors"
          v-else
          :key="`${uuid}-error-${error}`"
          class="input__error text-error-400">
          {{ error }}
        </span>
      </div>
    </slot>
  </ValidationProvider>
</template>
<script>
import { ValidationProvider } from 'vee-validate'
export default {
  name: 'InputValidator',

  components: {
    ValidationProvider
  },

  inheritAttrs: false,

  props: {
    bails: {
      type: Boolean,
      default: true
    },

    customMessages: {
      type: Object,
      default: () => ({})
    },

    debounce: {
      type: Number,
      default: 0
    },

    disabled: {
      type: Boolean,
      default: false
    },

    immediate: {
      type: Boolean,
      default: false
    },

    mode: {
      type: String,
      default: 'eager',
      validator: value =>
        [
          'aggressive',
          'passive',
          'lazy',
          'eager'
        ].includes(value)
    },

    name: {
      type: String,
      default: ''
    },

    rules: {
      type: [ String, Object ],
      default: null
    },

    skipIfEmpty: {
      type: Boolean,
      default: true
    },

    tag: {
      type: String,
      default: 'div'
    },

    vid: {
      type: String,
      default: ''
    },

    labelFor: {
      type: String,
      default: undefined
    },

    label: {
      type: String,
      default: undefined
    },

    hint: {
      type: String,
      default: ''
    },

    showErrors: {
      type: Boolean,
      default: true
    }
  }
}
</script>
<style lang="scss">
.input__errors,
.input__hint-container {
  margin-top: 0.4rem;
}

.input__error {
  font-size: 1.2rem;
  font-weight: 500;
  display: block;
  font-family: inherit;
}

.input__hint {
  display: block;
  font-size: 1.2rem;
  font-family: inherit;
}
</style>
