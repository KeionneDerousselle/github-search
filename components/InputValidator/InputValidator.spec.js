import flushPromises from 'flush-promises'
import { ValidationProvider, extend } from 'vee-validate'
import { required, min } from 'vee-validate/dist/rules'
import InputValidator from '.'

describe('InputValidator', () => {
  let wrapper
  const requiredValidationErrorMessage = '{_field_} is required.'
  const minValidationErrorMessage = '{_field_} cannot be less than {length} characters.'

  beforeAll(() => {
    extend('required', { ...required, message: requiredValidationErrorMessage })
    extend('min', { ...min, message: minValidationErrorMessage })
  })

  describe('when using the input validator', () => {
    let validationProvider

    beforeAll(() => {
      wrapper = mountPreMocked(InputValidator, {
        propsData: {
          name: 'myInputField'
        }
      })

      validationProvider = wrapper.findComponent(ValidationProvider)
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it("should pass vid prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().vid).toBe(wrapper.vm.vid)
    })

    it("should pass tag prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().tag).toBe(wrapper.vm.tag)
    })

    it("should pass skip-if-empty prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().skipIfEmpty).toBe(wrapper.vm.skipIfEmpty)
    })

    it("should pass rule prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().rules).toBe(wrapper.vm.rules)
    })

    it("should pass name prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().name).toBe(wrapper.vm.name)
    })

    it("should pass mode prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().mode).toBe(wrapper.vm.mode)
    })

    it("should pass immediate prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().immediate).toBe(wrapper.vm.immediate)
    })

    it("should pass disabled prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().disabled).toBe(wrapper.vm.disabled)
    })

    it("should pass debounce prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().debounce).toBe(wrapper.vm.debounce)
    })

    it("should pass custom-messages prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().customMessages).toBe(wrapper.vm.customMessages)
    })

    it("should pass bails prop to vee-validate's ValidationProvider", () => {
      expect(validationProvider.props().bails).toBe(wrapper.vm.bails)
    })
  })

  describe('when passing a label string to the input validator', () => {
    let labelText

    beforeAll(() => {
      labelText = 'myLabel'

      wrapper = mountPreMocked(InputValidator, {
        propsData: {
          name: 'myInputField',
          label: labelText
        }
      })
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should display the label', () => {
      expect(wrapper.get('.input__label').text()).toContain(labelText)
    })

    it('should set the screen reader only css class', () => {
      expect(wrapper.get('.input__label').classes()).toContain('sr-only')
    })
  })

  describe('when no label string is provided to the input validator', () => {
    beforeAll(() => {
      wrapper = mountPreMocked(InputValidator, {
        propsData: {
          name: 'myInputField'
        }
      })
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should not display the default label', () => {
      expect(() => wrapper.get('.input__label')).toThrow()
    })
  })

  describe('when there are no validation errors', () => {
    beforeAll(async () => {
      const Component = {
        components: {
          InputValidator
        },

        template: `
          <InputValidator rules="required" label="My Label">
            <input v-model="text"/>
          </InputValidator>
        `,

        data() {
          return {
            text: ''
          }
        }
      }

      wrapper = mountPreMocked(Component)

      wrapper.get('input').setValue('abc')
      await flushPromises()
      await wrapper.vm.$nextTick()

      wrapper.get('input').trigger('blur')
      await flushPromises()
      await wrapper.vm.$nextTick()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should not display the errors container', () => {
      expect(() => wrapper.get('.input__errors')).toThrow()
    })

    it('should not display any errors', () => {
      expect(() => wrapper.get('.input__error')).toThrow()
    })
  })

  describe('when there are validation errors, but the show errors property is false', () => {
    const inputId = 'my-input'
    const label = 'My Label'

    beforeAll(async () => {
      const Component = {
        components: {
          InputValidator
        },

        template: `
          <InputValidator rules="required" label="${label}" bails :show-errors="false">
            <input id="${inputId}" v-model="text"/>
          </InputValidator>
        `,

        data() {
          return {
            text: ''
          }
        }
      }

      wrapper = mountPreMocked(Component)

      wrapper.get('input').setValue('')
      await flushPromises()
      await wrapper.vm.$nextTick()

      wrapper.get('input').trigger('blur')
      await flushPromises()
      await wrapper.vm.$nextTick()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should not display the errors container', () => {
      expect(() => wrapper.get('.input__errors')).toThrow()
    })

    it('should not display any error messages', () => {
      expect(() => wrapper.get('.input__error')).toThrow()
    })
  })

  describe('when there are validation errors and the input validator should bail', () => {
    const inputId = 'my-input'
    const label = 'My Label'

    beforeAll(async () => {
      const Component = {
        components: {
          InputValidator
        },

        template: `
          <InputValidator rules="required" label="${label}" bails>
            <input id="${inputId}" v-model="text"/>
          </InputValidator>
        `,

        data() {
          return {
            text: ''
          }
        }
      }

      wrapper = mountPreMocked(Component)

      wrapper.get('input').setValue('')
      await flushPromises()
      await wrapper.vm.$nextTick()

      wrapper.get('input').trigger('blur')
      await flushPromises()
      await wrapper.vm.$nextTick()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should display the errors container', () => {
      expect(wrapper.get('.input__errors').element).toBeVisible()
    })

    it('should display a single error', () => {
      expect(wrapper.get('.input__error').element).toBeVisible()
    })

    it('should set the color of the error to text-error-400', () => {
      expect(wrapper.get('.input__error').classes()).toContain('text-error-400')
    })

    it('should display the error message', () => {
      expect(wrapper.get('.input__error').text()).toContain(requiredValidationErrorMessage.replace('{_field_}', label))
    })
  })

  describe('when there are validation errors and the input validator should not bail after the first error', () => {
    const inputId = 'my-input'
    const label = 'My Label'
    const length = 3

    beforeAll(async () => {
      const Component = {
        components: {
          InputValidator
        },

        template: `
          <InputValidator rules="required|min:${length}" label="${label}" :bails="false">
            <input id="${inputId}" v-model="text"/>
          </InputValidator>
        `,

        data() {
          return {
            text: ''
          }
        }
      }

      wrapper = mountPreMocked(Component)

      wrapper.get('input').setValue('')
      await flushPromises()
      await wrapper.vm.$nextTick()

      wrapper.get('input').trigger('blur')
      await flushPromises()
      await wrapper.vm.$nextTick()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should display the errors container', () => {
      expect(wrapper.get('.input__errors').element).toBeVisible()
    })

    it('should display the first error', () => {
      expect(wrapper.get('.input__error').element).toBeVisible()
    })

    it('should display the second error ', () => {
      expect(wrapper.get('.input__error + .input__error').element).toBeVisible()
    })

    it('should set the color of the first error to text-error-400', () => {
      expect(wrapper.get('.input__error').classes()).toContain('text-error-400')
    })

    it('should set the color of the second error to text-error-400', () => {
      expect(wrapper.get('.input__error + .input__error').classes()).toContain('text-error-400')
    })

    it('should display the first error message', () => {
      expect(wrapper.get('.input__error').text()).toContain(requiredValidationErrorMessage.replace('{_field_}', label))
    })

    it('should display the second error message', () => {
      expect(wrapper.get('.input__error + .input__error').text()).toContain(minValidationErrorMessage.replace('{_field_}', label).replace('{length}', length))
    })
  })

  describe('when a custom error slot is provided', () => {
    const inputId = 'my-input'
    const customErrorsSlotId = 'custom-error-message-slot'
    const customErrorId = 'custom-error-message'

    beforeAll(async () => {
      const Component = {
        components: {
          InputValidator
        },

        template: `
          <InputValidator rules="required" label="My Label">
            <input id="${inputId}" v-model="text"/>
            <div slot="errors" id="${customErrorsSlotId}">
              <span id="${customErrorId}">Uh oh! Please provide valid input.</span>
            </div>
          </InputValidator>
        `,

        data() {
          return {
            text: ''
          }
        }
      }

      wrapper = mountPreMocked(Component)

      wrapper.get('input').setValue('')
      await flushPromises()
      await wrapper.vm.$nextTick()

      wrapper.get('input').trigger('blur')
      await flushPromises()
      await wrapper.vm.$nextTick()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should display the custom error slot', () => {
      expect(wrapper.get(`#${customErrorsSlotId}`).element).toBeVisible()
    })

    it('should display the custom error message', () => {
      expect(wrapper.get(`#${customErrorId}`).element).toBeVisible()
    })

    it('should not display the default error messages container', () => {
      expect(() => wrapper.get('.input__errors')).toThrow()
    })

    it('should not display the default error message', () => {
      expect(() => wrapper.get('.input__error')).toThrow()
    })
  })

  describe('when a validator name is not provided', () => {
    const inputId = 'my-input'
    const inputLabel = 'My Input Field'

    beforeAll(async () => {
      const Component = {
        components: {
          InputValidator
        },

        template: `
          <InputValidator rules="required" label="${inputLabel}">
            <input id="${inputId}" v-model="text"/>
          </InputValidator>
        `,

        data() {
          return {
            text: ''
          }
        }
      }

      wrapper = mountPreMocked(Component)

      wrapper.get('input').setValue('')
      await flushPromises()
      await wrapper.vm.$nextTick()

      wrapper.get('input').trigger('blur')
      await flushPromises()
      await wrapper.vm.$nextTick()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should use the label in the displayed error message', () => {
      expect(wrapper.get('.input__error').text()).toContain(requiredValidationErrorMessage.replace('{_field_}', inputLabel))
    })
  })

  describe('when a validator name is provided', () => {
    const inputId = 'my-input'
    const validatorName = 'Some Custom Message'

    beforeAll(async () => {
      const Component = {
        components: {
          InputValidator
        },

        template: `
          <InputValidator rules="required" name="${validatorName}">
            <input id="${inputId}" v-model="text"/>
          </InputValidator>
        `,

        data() {
          return {
            text: ''
          }
        }
      }

      wrapper = mountPreMocked(Component)

      wrapper.get('input').setValue('')
      await flushPromises()
      await wrapper.vm.$nextTick()

      wrapper.get('input').trigger('blur')
      await flushPromises()
      await wrapper.vm.$nextTick()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should use the label in the displayed error message', () => {
      expect(wrapper.get('.input__error').text()).toContain(requiredValidationErrorMessage.replace('{_field_}', validatorName))
    })
  })

  describe('when a hint string is provided', () => {
    let hintText

    beforeAll(() => {
      hintText = 'This is a hint.'

      wrapper = mountPreMocked(InputValidator, {
        propsData: {
          name: 'myInputField',
          hint: hintText
        }
      })
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should display the hint', () => {
      expect(wrapper.get('.input__hint').text()).toContain(hintText)
    })

    it('should set the color of the hint to text black 500', () => {
      expect(wrapper.get('.input__hint').classes()).toContain('text-black-500')
    })
  })

  describe('when a hint slot is provided', () => {
    const inputId = 'my-input'
    const customHintsSlotId = 'custom-hints-slot'
    const customHintId = 'custom-hint'

    beforeAll(() => {
      const Component = {
        components: {
          InputValidator
        },

        template: `
          <InputValidator label="My Label">
            <input id="${inputId}" v-model="text"/>
            <div slot="hint" id="${customHintsSlotId}">
              <span id="${customHintId}">This is a hint.</span>
            </div>
          </InputValidator>
        `,

        data() {
          return {
            text: ''
          }
        }
      }

      wrapper = mountPreMocked(Component)
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should display the custom hints slot', () => {
      expect(wrapper.get(`#${customHintsSlotId}`).element).toBeVisible()
    })

    it('should display the custom hint', () => {
      expect(wrapper.get(`#${customHintId}`).element).toBeVisible()
    })

    it('should not display the default hint container', () => {
      expect(() => wrapper.get('.input__hint-container')).toThrow()
    })

    it('should not display the default hint', () => {
      expect(() => wrapper.get('.input__hint')).toThrow()
    })
  })

  describe('when a hint string is provided, but there is a validation error', () => {
    const inputId = 'my-input'
    const label = 'My Label'
    let hintText

    beforeAll(async () => {
      hintText = 'This is a hint.'

      const Component = {
        components: {
          InputValidator
        },

        template: `
          <InputValidator label="${label}" hint="${hintText}" rules="required">
            <input id="${inputId}" v-model="text"/>
          </InputValidator>
        `,

        data() {
          return {
            text: ''
          }
        }
      }

      wrapper = mountPreMocked(Component)

      wrapper.get('input').setValue('')
      await flushPromises()
      await wrapper.vm.$nextTick()

      wrapper.get('input').trigger('blur')
      await flushPromises()
      await wrapper.vm.$nextTick()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should not display the hints container', () => {
      expect(() => wrapper.get('.input__hint-container')).toThrow()
    })

    it('should not display the hint', () => {
      expect(() => wrapper.get('.input__hint')).toThrow()
    })

    it('should display the errors container', () => {
      expect(wrapper.get('.input__errors').element).toBeVisible()
    })

    it('should display a single error', () => {
      expect(wrapper.get('.input__error').element).toBeVisible()
    })

    it('should set the color of the error to text-error-400', () => {
      expect(wrapper.get('.input__error').classes()).toContain('text-error-400')
    })

    it('should display the error message', () => {
      expect(wrapper.get('.input__error').text()).toContain(requiredValidationErrorMessage.replace('{_field_}', label))
    })
  })

  describe('when a hint slot is provided, but there is a validation error', () => {
    const inputId = 'my-input'
    const customHintsSlotId = 'custom-hints-slot'
    const customHintId = 'custom-hint'
    const label = 'My Label'

    beforeAll(async () => {
      const Component = {
        components: {
          InputValidator
        },

        template: `
          <InputValidator label="${label}" rules="required">
            <input id="${inputId}" v-model="text"/>
            <div slot="hint" id="${customHintsSlotId}">
              <span id="${customHintId}">This is a hint.</span>
            </div>
          </InputValidator>
        `,

        data() {
          return {
            text: ''
          }
        }
      }

      wrapper = mountPreMocked(Component)

      wrapper.get('input').setValue('')
      await flushPromises()
      await wrapper.vm.$nextTick()

      wrapper.get('input').trigger('blur')
      await flushPromises()
      await wrapper.vm.$nextTick()
    })

    afterAll(() => {
      wrapper.destroy()
    })
    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should not display the custom hints slot', () => {
      expect(() => wrapper.get(`#${customHintsSlotId}`)).toThrow()
    })

    it('should not display the custom hint', () => {
      expect(() => wrapper.get(`#${customHintId}`)).toThrow()
    })

    it('should not display the default hint container', () => {
      expect(() => wrapper.get('.input__hint-container')).toThrow()
    })

    it('should not display the default hint', () => {
      expect(() => wrapper.get('.input__hint')).toThrow()
    })

    it('should display the errors container', () => {
      expect(wrapper.get('.input__errors').element).toBeVisible()
    })

    it('should display a single error', () => {
      expect(wrapper.get('.input__error').element).toBeVisible()
    })

    it('should set the color of the error to text-error-400', () => {
      expect(wrapper.get('.input__error').classes()).toContain('text-error-400')
    })

    it('should display the error message', () => {
      expect(wrapper.get('.input__error').text()).toContain(requiredValidationErrorMessage.replace('{_field_}', label))
    })
  })
})
