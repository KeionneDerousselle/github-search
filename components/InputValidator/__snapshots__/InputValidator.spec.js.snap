// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputValidator when a custom error slot is provided should render as expected 1`] = `
<div class="input"><label class="input__label sr-only">
    My Label
  </label> <input id="my-input">
  <!---->
  <div id="custom-error-message-slot"><span id="custom-error-message">Uh oh! Please provide valid input.</span></div>
</div>
`;

exports[`InputValidator when a hint slot is provided should render as expected 1`] = `
<div class="input"><label class="input__label sr-only">
    My Label
  </label> <input id="my-input">
  <div id="custom-hints-slot"><span id="custom-hint">This is a hint.</span></div>
  <!---->
</div>
`;

exports[`InputValidator when a hint slot is provided, but there is a validation error should render as expected 1`] = `
<div class="input"><label class="input__label sr-only">
    My Label
  </label> <input id="my-input">
  <!---->
  <div class="input__errors"><span class="input__error text-error-400">
        My Label is required.
      </span></div>
</div>
`;

exports[`InputValidator when a hint string is provided should render as expected 1`] = `
<div class="input">
  <!---->
  <div class="input__hint-container"><span class="input__hint text-black-500">This is a hint.</span></div>
  <!---->
</div>
`;

exports[`InputValidator when a hint string is provided, but there is a validation error should render as expected 1`] = `
<div class="input"><label class="input__label sr-only">
    My Label
  </label> <input id="my-input">
  <!---->
  <div class="input__errors"><span class="input__error text-error-400">
        My Label is required.
      </span></div>
</div>
`;

exports[`InputValidator when a validator name is not provided should render as expected 1`] = `
<div class="input"><label class="input__label sr-only">
    My Input Field
  </label> <input id="my-input">
  <!---->
  <div class="input__errors"><span class="input__error text-error-400">
        My Input Field is required.
      </span></div>
</div>
`;

exports[`InputValidator when a validator name is provided should render as expected 1`] = `
<div class="input">
  <!----> <input id="my-input">
  <!---->
  <div class="input__errors"><span class="input__error text-error-400">
        Some Custom Message is required.
      </span></div>
</div>
`;

exports[`InputValidator when no label string is provided to the input validator should render as expected 1`] = `
<div class="input">
  <!---->
  <!---->
  <!---->
</div>
`;

exports[`InputValidator when passing a label string to the input validator should render as expected 1`] = `
<div class="input"><label class="input__label sr-only">
    myLabel
  </label>
  <!---->
  <!---->
</div>
`;

exports[`InputValidator when there are no validation errors should render as expected 1`] = `
<div class="input"><label class="input__label sr-only">
    My Label
  </label> <input>
  <!---->
  <!---->
</div>
`;

exports[`InputValidator when there are validation errors and the input validator should bail should render as expected 1`] = `
<div class="input"><label class="input__label sr-only">
    My Label
  </label> <input id="my-input">
  <!---->
  <div class="input__errors"><span class="input__error text-error-400">
        My Label is required.
      </span></div>
</div>
`;

exports[`InputValidator when there are validation errors and the input validator should not bail after the first error should render as expected 1`] = `
<div class="input"><label class="input__label sr-only">
    My Label
  </label> <input id="my-input">
  <!---->
  <div class="input__errors"><span class="input__error text-error-400">
        My Label is required.
      </span><span class="input__error text-error-400">
        My Label cannot be less than 3 characters.
      </span></div>
</div>
`;

exports[`InputValidator when there are validation errors, but the show errors property is false should render as expected 1`] = `
<div class="input"><label class="input__label sr-only">
    My Label
  </label> <input id="my-input">
  <!---->
  <!---->
</div>
`;
